### DISCUSSION TITLE
Should i use useFieldArray for array of strings?

### DISCUSSION THREAD

**Discussion by hackerghost93** [react-hook-form/react-hook-form#5462] [Q&A] [answered]
I have an object { x: string[] }  and i want to make a number of inputs for it. should i use string[]. I don't think it is supported because when i tried it made an array of letters in each string. Also it only support array of objects. so what is the best practice here?

**Comment by bluebill1049** [comment]
I think a controlled component hosting all those inputs with `Controller` is the preferred and simple solution.

**Reply by hackerghost93** [reply]
Do you have any example for that?

**Reply by cbdeveloper** [reply]
@bluebill1049  would it be something like this?

```
// type FormState = { array: string[] }
// control: Control<FormState>;

// Array Inputs Component

<Controller control={control} name="array" render={({ field }) => {
  const array = field.value;  
  const { onChange } = field;           // On array level        
  const onChangeItem = (e: ChangeEvent<HTMLInputElement>, index: number) => {   // On array item level
    const newArray = deepClone(array);
    newArray[index] = e.target.value;
    onChange(e) // <-------- HOW TO CALL THIS onChange with Array<string> ?
  }                      
  const arrayInputs = array.map((item, index) => {
    return (
      <Input value={item} onChange={(e) => onChangeItem(e,index)} />
    );
  });
  return arrayInputs;
}}/>
```

How do I call the `field.onChange` with the updated array value?

**Reply by cbdeveloper** [reply]
I ended up implementing it with the `watch` value and `setValue`. Is it not recommended for some reason?

**Comment by thanh-nguyen-95** [comment, marked as answer]
You can either have a custom `TagInput` in combination with `Controller` or transform your data into `{ x: { value: string }[] }` and use `useFieldArray`

**Reply by hackerghost93** [reply]
Controller has name `tag.${number}` ?

**Reply by thanh-nguyen-95** [reply]
This is an example of `TagInput`: https://codeburst.io/building-a-tags-input-in-reactjs-2b59e8d3144a

After implementing the component, use it with `Controller`:

```jsx
<Controller
  control={control}
  name="x"
  render={({ field }) => (
    <TagInput {...field} /> // this depends on how you implement the component, change it however you want
  )}
/>
```
